src/tests/ut/medicine/medicine_test.cpp

```diff
@@ -4,14 +4,50 @@
 
 struct MedicineTest : ::testing::Test
 {
-    std::shared_ptr<Product> anaesthetic = std::make_shared<Medicine>(
-        std::string{"anaesthetic"}, 45, 100, 10122025, std::pair{-10, 0}, std::vector<std::string>{{"chemicals"}});
+    std::chrono::year_month_day expirationDate{static_cast<std::chrono::year>(2030), std::chrono::January,
+                                               static_cast<std::chrono::day>(31)};
+    std::shared_ptr<Product> anaesthetic =
+        std::make_shared<Medicine>(std::string{"anaesthetic"}, 45, 100, expirationDate, std::pair{-10, 0},
```

Comment:
Please switch struct to class to follow general apprach.
Please name magic numbers such 45,100, std::pair{-10,0}.
For that I suggest using constexpr above class. 
Ex. `constexpr int price{45};
class...`

Reply:
Done